t/lib/WTSI/NPG/Data/ConsentWithdrawnTest.pm

```diff
@@ -0,0 +1,148 @@
+package WTSI::NPG::Data::ConsentWithdrawnTest;
+
+use strict;
+use warnings;
+use English qw(-no_match_vars);
+use Test::More;
+use Test::Exception;
+use File::Temp qw( tempdir );
+use Log::Log4perl;
+use Test::MockObject;
+
+use WTSI::NPG::iRODS;
+
+use base 'WTSI::NPG::HTS::Test';
+
+Log::Log4perl::init('./etc/log4perl_tests.conf');
+
+my $pid          = $PID;
+my $test_counter = 0;
+my $collection;
+
+sub setup_test : Test(setup) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  $collection =
+    $irods->add_collection("RunPublisherTest.$pid.$test_counter");
+  $test_counter++;
+}
+
+sub teardown_test : Test(teardown) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  $irods->remove_collection($collection);
+}
+
+sub object_and_query_creation : Test(6) {
+
+  use_ok 'WTSI::NPG::Data::ConsentWithdrawn';
+
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  my $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods => $irods);
+  isa_ok($cw, 'WTSI::NPG::Data::ConsentWithdrawn');
+
+  throws_ok { WTSI::NPG::Data::ConsentWithdrawn->new() }
+    qr/Attribute \(irods\) is required/,
+    'constructor should have irods attr. defined';
+
+  my $query = q{"%s/%s" "select COLL_NAME, DATA_NAME where } .
+    q{META_DATA_ATTR_NAME = 'sample_consent_withdrawn' and } .
+    q{META_DATA_ATTR_VALUE = '1' and DATA_NAME not like '%header.bam%'"};
+
+  is($cw->_iquery, q{iquest --no-page } . $query,
+    'default iquery, no zone');
+
+  $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods => $irods);
+  is($cw->_iquery, q{iquest --no-page } . $query,
+    'iquery with zone set');
+
+  $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods      => $irods,
+                                               zone       => 'seq',
+                                               collection => '/some/other');
+  is($cw->_iquery, q{iquest --no-page -z seq } . substr($query, 0, -1) .q{ and COLL_NAME like '/some/other%'"},
+    'iquery with both zone and collection set');
+}
+
+sub permissions : Test(21) {
+
+  my $mock = Test::MockObject->new();
+  $mock->fake_new( 'MIME::Lite' );
+  $mock->set_true('send');
```

Comment:
Is the mock object being used? I can't see it referenced elsewhere.

Reply:
No, it is not. It is defined in these lines. When a new MIME::Lite object is created by the code, it's this instance that appears there.

This is printing the object in the test:
# My mock Test::MockObject=HASH(0x4d11518)

This is printing a new MIME::Lite object  in the code:
# IN CODE Test::MockObject=HASH(0x4d11518)